### DISCUSSION TITLE
Leox LXT-010S-H vlan problem

### DISCUSSION THREAD

**Discussion by keczejo** [Anime4000/RTL960x#354] [Help]
I bought leox LXT-010S-H as a replacement for ISP provided Zhone ZNID-GPON-2424-EU. I changed data to immitate Zhone device and i get O5 gpon state. My ISP is using VEIP and on top of that DHCP for address provisioning. Also i have multicast IPTV on a separate VLAN.

![image (1).png](https://github.com/Anime4000/RTL960x/assets/61348735/802bcd82-12f3-46a1-b652-3a1042963b9a)

![image (2).png](https://github.com/Anime4000/RTL960x/assets/61348735/13631c0a-ed5d-49a4-8800-f02fc309ee74)

![image (3).png](https://github.com/Anime4000/RTL960x/assets/61348735/732fed3a-3c0a-4a38-81c7-4e5988f2ec0e)

VLANs:
246 multicast IPTV
725 DHCP

At the first i didn't get any data comming from sfp interface. 
I contacted with Leox support and they provided me with firmware supporting VEIP. After firmware upgrade i started to recieve around 100Mb/s of data on sfp interface. It was multicast traffic on vlan246. I changed mac address on the router and enabled dhcp client on vlan 725, but there was no response. I also tried to bridge vlan 246 with ethernet port connected to STB. Unfortunately there was no television. 

Looking on PON statistics i see increasing amount of recieved data but number of data sent is 0 even though i am sending data to a proper VLAN on sfp interface. 

![Screenshot_20240126-125443_Chrome.png](https://github.com/Anime4000/RTL960x/assets/61348735/2b84b01b-af85-4419-b836-01cc9856b4c9)

![Screenshot_20240126-153131_JuiceSSH.png](https://github.com/Anime4000/RTL960x/assets/61348735/ff4b7849-e099-403e-b940-1b6140515846)

![image (6).png](https://github.com/Anime4000/RTL960x/assets/61348735/e35c413d-3c0c-4769-bcd2-d502406169c1)

![Screenshot_20240228-144703_Gmail.png](https://github.com/Anime4000/RTL960x/assets/61348735/ab7fef49-6e0f-4fd7-9af3-2d74931bcc01)

![image (5).png](https://github.com/Anime4000/RTL960x/assets/61348735/45b8426b-0c18-4f4e-8247-ec6b596d3392)

![image (4).png](https://github.com/Anime4000/RTL960x/assets/61348735/6fcfba4c-3d6b-42bf-9e54-6873b96b6d6f)

I tried this command `omcicli mib set 84 0x02 FwdOp 0x02 but it haven't changed anything.`

**Comment by vuducdong** [comment]
Firmware that supports VEIP will have a V at the end, but I see your photo above is only V3.3.4L4. It should have been V3.3.4LV. Is there a mistake here or did Leox change the naming rules?

**Comment by keczejo** [comment]
> 
> Firmware that supports VEIP will have a V at the end, but I see your photo above is only V3.3.4L4. It should have been V3.3.4LV. Is there a mistake here or did Leox change the naming rules?

My bad 
This screenshot was taken with firmware V3.3.4L4. On this software version there was no data on sfp port. I recieved firmware V3.3.4L5VT and V3.3.4L5V from leox support. On V3.3.4L5V OMCI_OLT_MODE=3 wasn't working, on V3.3.4L5VT OMCI_OLT_MODE=3 was working, i was receiving data on vlan 246, but data send by my router wasn't shown in pon statistics, exactly like on the screen shot.

ISP - > router is working
Router - > ISP isn't working

**Comment by ToTheCLI** [comment]
DHCP (IPOE) sometimes needs MAC Address Cloning on SFP Interface, try to clone the MAC Address of original ONT WAN Interface.

**Comment by keczejo** [comment]
> DHCP (IPOE) sometimes needs MAC Address Cloning on SFP Interface, try to clone the MAC Address of original ONT WAN Interface. 

I did that as a first thing. Weird thing is that my sfp module recieves DHCP requests but don't forward them to the optical interface.

**Comment by vuducdong** [comment]
> > DHCP (IPOE) sometimes needs MAC Address Cloning on SFP Interface, try to clone the MAC Address of original ONT WAN Interface.
> 
> I did that as a first thing. Weird thing is that my sfp module recieves DHCP requests but don't forward them to the optical interface.

You can try it at the same time:
1, clone MAC ONT ---> MAC SFP GPON (Leox)
2, clone MAC port WAN Router (before running) ---> MAC port WAN router (new) (in this case, the sfp port)

**Comment by BartonSVK** [comment]
I'm gonna watch this thread closely as I've got my stick yesterday, installed V3.3.4L5 and it doesn't work too :) I wonder if I need V or VT too (what's the difference between them anyway?) for Orange Slovakia (they use different VLAN's for INT, TV and VOIP too)...

**Comment by keczejo** [comment]
> > > DHCP (IPOE) sometimes needs MAC Address Cloning on SFP Interface, try to clone the MAC Address of original ONT WAN Interface.
> > 
> > I did that as a first thing. Weird thing is that my sfp module recieves DHCP requests but don't forward them to the optical interface.
> 
> You can try it at the same time:
> 1, clone MAC ONT ---> MAC SFP GPON (Leox)
> 2, clone MAC port WAN Router (before running) ---> MAC port WAN router (new) (in this case, the sfp port)

Which command should i use to change ONT MAC?
I don't think it is the problem. I am sending requests and looks like the request is not send by the module to the OLT (bytes send counter is 0). I think sfp module doesn't permit packets from sfp to pon interface.

ROUTER -> SFP works 
SFP -> PON doesn't work

**Comment by Anime4000** [comment]
have you tried to change FwdOp to 0x02 ? (Bridge, no investigation)

**Comment by keczejo** [comment]
> have you tried to change FwdOp to 0x02 ? (Bridge, no investigation)

Yes i tried this solution. There was no change to the traffic. Should i do anything more after typing this command, like replugging fiber or something?

**Comment by Anime4000** [comment]
it should work immediately, have you tried to conect Loex?

**Comment by keczejo** [comment]
That was the first thing i did. They provided me different firmware versions and to try `omcicli mib set 84 0x02 FwdOp 0x02` . For two weeks i haven't received a message from them. 

Can I try to install firmware from different SFP module with the same SoC on this one?

**Reply by belkone** [reply]
Hello @keczejo did you finally manage to do anything about it?

**Comment by Anime4000** [comment]
RTL9601CI = V2801F, TWCGPON657 and many others, yea, but Leox stick is build differently even using same SoC, I would not recommend to flash it

if want to test, I advised you dump/backup whole SPI Flash, then you can try flash different fw

**Comment by Mr1070** [comment]
i have had great luck with LeoLabs SFP .... i followed https://medium.com/@cyayon/configure-ont-leox-lxt-010h-d-for-orange-isp-4ef4b793ca7b and also took the advice to "flash set" the very minimum it is worked great

**Comment by umalavasic** [comment]
I have the same problem. LXT-010S-H on `V3.3.4L5V`, Iskratel OLT (T-2 Slovenia).
O5, downstream multicast works, but upstream `Bytes Sent` is stuck at 0.

Frames from my router reach the stick fine, but they're never handed to the fibre side:
- The stick learns my router's MAC on the host/SFP port and the ISP's MACs on the PON side, both on the right VLAN. So tagging and the link are OK and both directions see traffic.
- But upstream, `gpon get bwmap` shows the OLT only ever giving transmit slots to the OMCI/management channel and a default one, never to the data T-CONT (the upload queue for GEM 669). And the stick reports it has nothing queued to send, so none of my router's traffic is actually getting into the upload queue.

The stick has no physical Ethernet UNI in its bridge, only the VEIP, so the firmware itself has to connect the SFP host port to the VEIP for upstream, and on this build it doesn't seem to.
